### ISSUE TITLE
[format] Set up line length & rules for Dart code when using VSCode

### ISSUE THREAD

**Issue by navaronbracke** [flutter/flutter#160734]
### Use case

Developers wanting to contribute to the framework (or engine), by writing Dart code, while having their VSCode `User` settings set to something that differs from the framework's settings.

Context:

In https://github.com/flutter/flutter/commit/5491c8c146441d3126aff91beaa3fb5df6d710d0 the settings for `formatOnSave` and `formatOnPaste` were removed, as those were explicitly set to `false`. This presumably restored VSCode's default settings to format by default.

I started work on fixing a bug in the framework, post the engine binaries outage from yesterday, having not worked on the framework since the auto format change landed. Since I was aware of the autoformatting change, I pulled from master at 11AM UTC+1 on 22/12/2024, to start without issues. However, when using the `Format Document` action in `carousel_test.dart` I had a formatting diff in a piece of code that I did not write.

However, I have the following in my `User` settings, which overrides the `Workspace` settings from VSCode, if it finds things that are not defined.

```
    "dart.lineLength": 120,
    "[dart]": {
        "editor.formatOnSave": true,
        "editor.formatOnType": true,
        "editor.rulers": [
            120
        ],
    },
```

Neither the `dart.lineLength` nor `editor.rulers` is defined by the framework's `settings.json`, so I think it will use my own (incorrect) settings?

### Proposal

Add 

```
    "dart.lineLength": frameworkLineLength,
    "[dart]": {
        "editor.formatOnSave": true,
        "editor.formatOnType": true,
        "editor.rulers": [
            frameworkLineLength
        ],
    },
```

to the framework's `.vscode/settings.json`, where `frameworkLineLength` is the desired value for formatting code in the framework. By making the line length & formatOnSave/Type explicit, we avoid this being overridden by user level settings.

**Comment by DanTup** [comment]
IMO, yes - the framework settings should be committed here because it shouldn't be assumed that users defaults are the same as Flutter uses.

However, all users might not open the same folders here, so putting them only in the root might not be enough if some users might just open the `flutter_tools` folder, for example.

(I use my own `.code-workspace` file so I can have only the projects I want, and with my own settings that don't have to be committed, but I don't know how common that is).

**Comment by jtmcdole** [comment]
If we have a requirement for the line length that differs, we should set them in the shared vscode config.  I've suggested having multiple "framework.code-workspace", "all.code-workspace", "engine.code-workspace" as well.

**Comment by goderbauer** [comment]
Yes, it makes sense to explicitly configure the options that we use. Please feel free to send a PR for this.

**Comment by DanTup** [comment]
I think this might be stale now? Page width can be set in analysis_options and is set for Flutter:

https://github.com/flutter/flutter/blob/db288c46666ede5dfadb4b2bddbd3f0785c08853/analysis_options.yaml#L40-41

That should override the `dart.lineLength` setting (which will be deprecated). If it's not, please let me know as I'd consider that a bug and would like to fix.

**Comment by navaronbracke** [comment]
I should check what happens if I remove that setting

**Comment by DanTup** [comment]
> I should check what happens if I remove that setting

For the Flutter repo (which has it in the analysis_options), you should get the correct behaviour (a page width that matches what's in analysis_optons) both with and without the VS Code setting. The `dart.lineLength` setting should _only_ be used when there isn't something in analysis_options.
